lib/DataSource.php

```diff
@@ -84,12 +84,19 @@ public function __construct(IQueryLogger $queryLogger, IEventLogger $eventLogger
 		$this->diagnostics = $diagnostics;
 	}
 
+	public function activateDataSources() {
+		if ($this->diagnostics->isDiagnosticActivatedForSession()) {
```

Comment:
isDiagnosticActivatedForSession() will check if user is allowed to activate loggin for event and query logger